pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -292,6 +289,11 @@ def simulate_and_vjp(
 
 def stopping_condition(op: Operator) -> bool:
     """A function that determines whether or not an operation is supported by ``lightning.qubit``."""
+    if isinstance(op, qml.QFT):
+        return len(op.wires) < 10
+    if isinstance(op, qml.GroverOperator):
+        return len(op.wires) < 13
+
```

Comment:
Could we split the logic in `lightning_base.py` so that we can reuse this here too?
https://github.com/PennyLaneAI/pennylane-lightning/blob/6d374fd9084ec804443532b3088599761603d1b3/pennylane_lightning/core/lightning_base.py#L96-L103

Reply:
Do you mean define `accepts_obj` somewhere such that every backend can use it?

Reply:
Currently `LightningQubit` is not inheriting from LightningBase.
Not after the migration to the new device API.

Reply:
I want us to ensure that we centralize these magic thresholds in one place for the sake of maintainability.

Reply:
I agree. We touched on this briefly in discussions when first adding the new device and I believe that our decision was to keep such helper functions in `lightning_qubit.py` for now. We can move them to a separate file for shared access once we move at least one other device to the new API.

Reply:
Adding these magical numbers in multiple places is not a good practice. As this is not the main objective of this PR, let's add a note here to highlight the adaption here. We can address it in a followup PR :)

Reply:
We should, at some point, allocate some time to create a new device API LightningBase.
This will centralize these numbers, as we had before, and serve as pre-work for migrating LightningKokkos and LightningGPU.